src/Elastic.Apm/DiagnosticSource/DiagnosticInitializer.cs

```diff
@@ -18,11 +24,24 @@ internal class DiagnosticInitializer : IObserver<DiagnosticListener>, IDisposabl
 
 		public void OnNext(DiagnosticListener value)
 		{
+			_logger.Trace()?.Log(nameof(OnNext) + " called with value.Name: {DiagnosticListenerName}", value.Name);
+
+			var subscribedAny = false;
 			foreach (var listener in _listeners)
 			{
 				if (value.Name == listener.Name)
+				{
 					_sourceSubscription = value.Subscribe(listener);
+					_logger.Debug()
+						?.Log("Subscribed {DiagnosticListenerType} to `{DiagnosticListenerName}' events source",
+							listener.GetType().FullName, value.Name);
+					subscribedAny = true;
+				}
 			}
+
+			if (!subscribedAny)
```

Comment:
I think this can be problematic, because this is called for every single subscriber. E.g. if you have HttpClient, EfCore, AspNetCore in the app then this runs for all 3, and if you want to subscribe to e.g. HttpClient then 2 will fail and that's normal. E.g. this is the output when I start `SampleAspNetCore` app:

```
[2019-04-25 09:30:26][Debug] - {HttpDiagnosticListener} Current runtime is not detected as Full Framework - returning implementation for Core. RuntimeInformation.FrameworkDescription: .NET Core 4.6.27129.04
[2019-04-25 09:30:26][Trace] - {DiagnosticInitializer} OnNext called with value.Name: Microsoft.AspNetCore
[2019-04-25 09:30:26][Trace] - {DiagnosticInitializer} There are no listeners that would like to subscribe to `Microsoft.AspNetCore' events source
[2019-04-25 09:30:26][Trace] - {DiagnosticInitializer} OnNext called with value.Name: Microsoft.AspNetCore
[2019-04-25 09:30:26][Trace] - {DiagnosticInitializer} There are no listeners that would like to subscribe to `Microsoft.AspNetCore' events source
[2019-04-25 09:30:26][Trace] - {DiagnosticInitializer} OnNext called with value.Name: Microsoft.AspNetCore
[2019-04-25 09:30:26][Debug] - {DiagnosticInitializer} Subscribed Elastic.Apm.AspNetCore.DiagnosticListener.AspNetCoreDiagnosticListener to `Microsoft.AspNetCore' events source
```

This happens when we try to subscribe to `Microsoft.AspNetCore` events. 

Then the same comes when we subscribe to EF Core, and HttpClient. I'd simply just log the successful subscribes (as you do above) and then based on those logs we can safely say which listeners are subscribed and which ones aren't .

Reply:
I thought it might be useful to know what other event sources are available - maybe we will see that some listener of ours did not subscribe but on the other hand there were some event sources with similar names. That is why it's on the TRACE level - it's doesn't signify an important event - just logging the flow. I do agree that it would be nicer to print it only once but on the other hand I assume we won't have too many listeners, will we?

Reply:
👍 As discussed we adapted the line a little bit, it looks ok now.